File: src/Settings/Readiness.php
<?php
declare(strict_types=1);
namespace App\Settings;
use App\Container;
final class Readiness
{
    public static function report(Container $app):array
    {
        $c=$app->config;$db=$app->db;$checks=[];
        $add=function(string $label,bool $ok,string $detail)use(&$checks){$checks[]=['label'=>$label,'ok'=>$ok,'detail'=>$detail];};
        $add('Боевой режим',$c['APP_ENV']==='prod','Включите после настройки HTTPS и интеграций.');
        $add('PostgreSQL',$db->postgres(),'SQLite подходит только для локальной разработки.');
        $add('Ограниченные права базы',\App\Infrastructure\Permissions::safe($db),'Процесс приложения не должен быть владельцем таблиц или изменять финансовый журнал.');
        $add('HTTPS кабинета',str_starts_with($c['APP_URL'],'https://'),'URL должен совпадать с публичным адресом.');
        $add('Реальные адаптеры',in_array($c['PAYMENT_DRIVER'],['platega'],true)&&$c['PROVISION_DRIVER']==='remnawave','Демо не выдаёт реальный доступ.');
        $add('Параметры оплаты и панели',Settings::purchaseErrors($c)===[],'Магазин, ключи и группа панели должны быть заполнены.');
        $paymentCheck='platega';
        foreach(['telegram',$paymentCheck,'remnawave'] as $name){$row=$db->one('SELECT * FROM integration_checks WHERE integration=?',[$name]);$ok=$row && $row['status']==='ok' && hash_equals($row['config_hash'],IntegrationCheck::fingerprint($c,$name)) && (int)$row['checked_at']>time()-86400;$add('Проверка '.$name,(bool)$ok,'Проверка текущих настроек должна пройти в последние 24 часа.');}
        $admins=$db->all("SELECT totp_secret FROM users WHERE role='admin' AND disabled=0");
        $add('Защита администраторов',count($admins)>0&&!in_array(null,array_column($admins,'totp_secret'),true),'Для каждого администратора обязательна 2FA.');
        foreach(['worker','scheduler'] as $name){$row=$db->one('SELECT seen_at FROM runtime_heartbeats WHERE name=?',[$name]);$add('Процесс '.$name,$row && (int)$row['seen_at']>time()-180,'Ожидается сигнал не старше трёх минут.');}
        $add('Очередь без остановленных заданий',$db->one("SELECT id FROM outbox WHERE status='dead' LIMIT 1")===null,'Исправьте причину и повторите задание.');
        $add('Доступные тарифы',$db->one('SELECT id FROM plans WHERE active=1 LIMIT 1')!==null,'Создайте хотя бы один активный тариф.');
        $add('Продажи включены',$c['PURCHASES_ENABLED']==='1','Включайте после проверки тестовой покупки и восстановления.');
        return $checks;
    }
}

File: ruby_app/zeleboba/settings/readiness.rb
# frozen_string_literal: true

module Zeleboba
  module Settings
    class Readiness
      def self.report(container)
        config = container.config
        now = Time.now.to_i
        checks = []
        add = ->(name, ok, detail) { checks << { "name" => name, "ok" => ok, "detail" => detail } }

        add.call("Production mode", config["APP_ENV"] == "prod", "Set APP_ENV=prod only after integration checks pass.")
        add.call("PostgreSQL", container.db.postgres?, "SQLite is development-only.")
        add.call("HTTPS", config["APP_URL"].to_s.start_with?("https://"), "APP_URL must be the public HTTPS address.")
        add.call("Payment provider", config["PAYMENT_DRIVER"] == "platega" && config["PLATEGA_ENABLED"] == "1", "Use Platega in production.")
        add.call("Payment credentials", !config["PLATEGA_MERCHANT_ID"].to_s.empty? && !config["PLATEGA_SECRET"].to_s.empty?, "Provide Platega merchant credentials.")
        add.call("Provisioning provider", config["PROVISION_DRIVER"] == "remnawave", "Demo provisioning is not production-safe.")
        add.call("MFA encryption", !config["MFA_ENCRYPTION_KEY"].to_s.empty?, "Provide a Base64 32-byte MFA_ENCRYPTION_KEY.")
        add.call("Worker queue", container.db.one("SELECT id FROM outbox WHERE status='dead' LIMIT 1").nil?, "Resolve dead-letter jobs before launch.")
        add.call("Plans", !container.db.one("SELECT id FROM plans WHERE active=1 LIMIT 1").nil?, "At least one active plan is required.")
        add.call("Workers", heartbeat_ok?(container, "worker", now) && heartbeat_ok?(container, "scheduler", now), "Worker and scheduler heartbeats must be newer than three minutes.") if table_exists?(container, "runtime_heartbeats")
        checks
      end

      def self.ready?(container)
        report(container).all? { |check| check["ok"] }
      end

      def self.heartbeat_ok?(container, name, now)
        container.db.one("SELECT seen_at FROM runtime_heartbeats WHERE name=?", [name])&.fetch("seen_at", 0).to_i > now - 180
      end
      private_class_method :heartbeat_ok?

      def self.table_exists?(container, name)
        if container.db.postgres?
          !!container.db.one("SELECT to_regclass(?) AS name", [name])&.fetch("name", nil)
        else
          !!container.db.one("SELECT name FROM sqlite_master WHERE type='table' AND name=?", [name])
        end
      end
      private_class_method :table_exists?
    end
  end
end
